go-controller/pkg/ovn/pods.go

```diff
@@ -238,31 +237,40 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 	var podMac net.HardwareAddr
 	var podIfAddrs []*net.IPNet
-	var args []string
-	var addresses string
+	var cmds []*goovn.OvnCommand
+	var addresses []string
+
+	// Check if the pod's logical switch port already exists. If it
+	// does don't re-add the port to OVN as this will change its
+	// UUID and and the port cache, address sets, and port groups
+	// will still have the old UUID.
+	lsp, err := oc.ovnNBClient.LSPGet(portName)
+	if err != nil && err != goovn.ErrorNotFound && err != goovn.ErrorSchema {
+		return fmt.Errorf("unable to get the lsp: %s from the nbdb: %s", portName, err)
+	}
+
+	if lsp == nil {
+		lspAddCmd, err := oc.ovnNBClient.LSPAdd(logicalSwitch, portName)
```

Comment:
@abhat I'd say we just name all these variables `cmd` (rather than unique names) since we're only using it immediately after the creation, but your call.